File: src/cn/edustar/jitar/util/HtmlPager.java
package cn.edustar.jitar.util;

public class HtmlPager {

    public static String render(int pageCount, int offsetCount, int current) {
        String preLabel = "<";
        String nextLabel = ">";
        String h = "";
        if (pageCount <= 1)
            return h;
        int MaxStart = pageCount - offsetCount * 2;
        if (MaxStart < 1)
            MaxStart = 1;
        int MinEnd = offsetCount * 2 + 1;

        int start = current - offsetCount;
        int end = current + offsetCount;
        if (start < 1) {
            start = 1;
            end = MinEnd;
        }
        if (end > pageCount) {
            end = pageCount;
            start = MaxStart;
        }
        boolean showGoInput = false;
        h = "<a href='javascript:void(0)' class='listPagePre'>" + preLabel + "</a> ";
        if (current == 1)
            h = "";
        if (start > 1)
            h += "<a href='javascript:void(0)' class='listPageC'>1</a> ";
        if (start > 2) {
            showGoInput = true;
            h += "<span>...</span>";
        }

        for (int i = start; i < end; i++) {
            if (i == current) {
                h += "<a href='javascript:void(0)' class='listPageC active'>" + i + "</a> ";
            } else {
                h += "<a href='javascript:void(0)' class='listPageC'>" + i + "</a> ";
            }
        }
        if (start < MaxStart) {
            showGoInput = true;
            h += "<span>...</span>";
        }
        if (current == pageCount) {
            h += " <a href='javascript:void(0)' class='listPageC active'>" + current + "</a>";
        } else {
            h += " <a href='javascript:void(0)' class='listPageC'>" + pageCount + "</a>";
        }
        if (current != pageCount) {
            h += "<a href='javascript:void(0)' class='listPagePre'>" + nextLabel + "</a> ";
        }
        if (showGoInput) {
            h += " <span class='listPageText'>跳转至</span><span class='listPageInput'><input minValue='1' maxValue='" + pageCount + "' value='" + current
                    + "' type='text'></span><span class='listPageBtn'><input type='button' value='GO'></span>";
        }
        return h;

    }
}

File: WebContent/js/jitar/htmlpager.js
var Class =window.Class || {
	create : function() {
		return function() {
			this.initialize.apply(this, arguments);
		};
	}
};
var Pager = Class.create();
Pager.prototype = {
	initialize : function(divId, pageCount, offsetCount, callBack) {
		this.pageCount = pageCount;
		this.offsetCount = offsetCount;
		this.callBack = callBack;
		this.current = 1;
		this.preLabel = "上一页";
		this.nextLabel = "下一页";
		this.divId = divId;
		obj = this;
	},
	transferTo:function(evt)
	{
		var ele = evt?evt.target:window.event.srcElement;
		ele = ele.previousSibling;
		while(ele.nodeType != 1) ele = ele.previousSibling;
		var goPage = parseInt(ele.value,10);
		if(isNaN(goPage)) goPage = 1;
		if(goPage<1) goPage=1;
		if(goPage>this.pageCount) goPage=this.pageCount;
		this.callBack(goPage);
		this.current = goPage;
		this.build();
	},
	build : function() {
		if (this.pageCount < 1)
			return;
		var MaxStart = this.pageCount - this.offsetCount * 2;
		if (MaxStart < 1)
			MaxStart = 1;
		var MinEnd = this.offsetCount * 2 + 1;

		var start = this.current - this.offsetCount;
		var end = this.current + this.offsetCount;
		if (start < 1) {
			start = 1;
			end = MinEnd;
		}
		if (end > this.pageCount) {
			end = this.pageCount;
			start = MaxStart;
		}
		var showGoInput = false;
		var h = "<a href='javascript:void(0)'>" + this.preLabel + "</a> ";
		if(this.current == 1) h = "";
		if (start > 1)
			h += "<a href='javascript:void(0)'>1</a> ";
		if (start > 2){
			showGoInput = true;
			h += "<span class='dot'>...</span>";
		}
			
		var lastNo = 0;
		for (i = start; i < end; i++) {

			if (i == this.current) {
				h += "<span class='current'>" + i + "</span> ";
			} else {
				h += "<a href='javascript:void(0)'>" + i + "</a> ";
			}
			lastNo = i;
		}
		if (start < MaxStart){
			showGoInput = true;
			h += "<span class='dot'>...</span>";
		}
		if (this.current == this.pageCount) {
			h += "<span class='current'>" + this.current + "</span> ";
		} else {
			h += " <a href='javascript:void(0)'>" + this.pageCount + "</a>";
		}
		if(this.current != this.pageCount){
			h += "<a href='javascript:void(0)'>" + this.nextLabel + "</a> ";
		}
		if(showGoInput){
			h += " <input size='4' class='transferinput'/> <input type='button' onclick='obj.transferTo(event)' value='转到' class='transferto' />";
		}
		document.getElementById(this.divId).innerHTML = h;
		var bar = document.getElementById(this.divId).getElementsByTagName("a");
		for (i = 0; i < bar.length; i++) {
			bar[i].onclick = function(evt) {
				var pageNo = parseInt(this.innerHTML, 10);
				if (isNaN(pageNo)) {
					if (this.innerHTML == obj.preLabel)
						pageNo = Math.max(1, obj.current - 1);
					else if (this.innerHTML == obj.nextLabel)
						pageNo = Math.min(obj.pageCount, obj.current + 1);
					else {
						pageNo = 1;
					}
					;
				}

				obj.callBack(pageNo);
				obj.current = pageNo;
				obj.build();
				if (!evt)
					var evt = window.event;
				evt.cancelBubble = true;
				if (evt.stopPropagation)
					evt.stopPropagation();
			};
		}
	}
};